### ISSUE TITLE
Merge fork-arnopaehler, take 2

### ISSUE THREAD

**Issue by ChALkeR** [qmlweb/qmlweb#163]
This consists of everything that is left in #17 minus the unwanted files and style-only changes.

First set of patches (92eb17d split): [fork-arnopaehler-patches1](https://github.com/qmlweb/qmlweb/commits/fork-arnopaehler-patches1). Commits:
- [x] 5252956 Implement logCss() function — **Rejected**, unused debug method.
- [x] a712120 Implement TextEdit — **Merged** in #174.
- [x] 5feb0bb Implement TestPlugin — **Rejected**, this is non-standard, looks like a playground. We don't need this.
- [x] 79c6678 .gitignore: Add .idea/ and /.project — **Merged** in #174.

Second set of patches (6830c351056f16cc4f9a57060e7a1ced7964d846 split):  [fork-arnopaehler-patches2](https://github.com/qmlweb/qmlweb/commits/fork-arnopaehler-patches2). Commits:
- [x] 2585cb5 Row: register effectiveLayoutDirection — **Rejected**, it is never set here (this is a stub), and this is not the place where should be registered — it should be present for other positioners too, e.g. ListView, Flow, Grid, etc. 
- [x] fb6cd39 Add Qt.Align* constants — **Merged** in #167.
- [x] e7e6523 Implement Screen — **Merged** in #167.
- [x] 384b250 parser: walker updates — **Rejected**, see https://github.com/qmlweb/qmlweb/issues/163#issuecomment-208813734. 
- [x] 0a09a8c More work on Loader — **Merged** in #167.
- [x] 41f545e A clarification of a TODO comment — **Rejected**, inapplicable. That hack for older browsers was removed.
- [x] ba8b9ff Register Array as 'array' basic QML type — **Rejected**, see #149.
- [x] 8fe2ad9 Implement TextField — **Merged** in #174.
- [x] 79e6124 TextArea: use strict comarison in updateValue — **Rejected**, inapplicable, and I don't see why we would want this. In @arnopaehler fork, TextEdit doesn't have this change.
- [x] 1bded10 Implement objItems() and typeName() — **Rejected**, unused debug methods.
- [x] e7cee0d Default to 'Object List' in objList() title argument — **Rejected**, unused debug methods.
- [x] 36f6b2b SequentialAnimation: remove debug, remove descr() debug function — **Already done**, in #29 (63f0a081a62136239305faf07e115d7193e863a3).
- [x] 501a197 Implement ComboBox — **Merged** in #167.
- [x] 3b9fe97 Item: use overflow: auto instead of overflow: hidden — **Rejected**, we don't want random scrollbars.
- [x] 954894b QMLEngine: directly assign e.keypad in keyCodeToQt(e) — Looks valid, though all that code needs better refactoring. **Merged** in #167.
- [x] 43bccf8 Move Button to QtQuick.Controls — **Already done**, in #122 (bce018c63436f7453a63990a293292a2ac7c4f10).
- [x] 55bd5e8 Implement Canvas — **Merged** in #167.
- [x] 8cdd241 [ignore] Remove a Canvas TODO stub — **Rejected**, this is a stub, inapplicable as we didn't have it.
- [x] c9def86 Add a test signal to TestPlugin — **Rejected**, as TestPlugin is rejected.

**Comment by ChALkeR** [comment]
I'm not sure what 384b250 does.

**Comment by ChALkeR** [comment]
I don't think we need 5252956, 1bded10, e7cee0d — those are debug methods that aren't used anywhere.

**Comment by Plaristote** [comment]
I've been through 384b250 before. I think we should reject it, it works around something that wasn't implemented. Back in those days, there was no walkers for the array, num and string types.

But now these walkers exists, they've been implemented, I believe, in kde-master by @akreuzkamp, and it's already merged in master.

I agree about the 4 other commits, we shouldn't merge them.

**Comment by ChALkeR** [comment]
@Plaristote Thanks for the clarification!

**Comment by ChALkeR** [comment]
I also marked 79e6124 as rejected, it's inapplicable, and I don't see why we would want this. In @arnopaehler fork, TextEdit doesn't have this change.

**Comment by ChALkeR** [comment]
I have not yet considered 41f545e, everything else should be in #174.

**Comment by ChALkeR** [comment]
Everything here got looked at, this can be closed once #174 lands.

**Comment by ChALkeR** [comment]
This is now merged.
